js/src/utils/external.ts

```diff
@@ -21,36 +21,48 @@ export async function sendProcessReq(info: {
     );
     return true;
   }
+
   try {
-    // Spawn a child process to execute a Node.js script
+    // Use node-fetch for making HTTP requests
+    const url = new URL(info.url);
     const child = spawn("node", [
       "-e",
       `
-        const http = require('http');
+        const http = require('${url.protocol === "https:" ? "https" : "http"}');
         const options = {
-            hostname: '${info.url}',
-            method: 'POST',
-            headers: ${JSON.stringify(info.headers)}
+          hostname: '${url.hostname}',
+          path: '${url.pathname}${url.search}',
+          port: ${url.port || (url.protocol === "https:" ? 443 : 80)},
+          method: '${info.method}',
+          headers: ${JSON.stringify(info.headers)}
         };
-        
+
         const req = http.request(options, (res) => {
-            console.log('statusCode:', res.statusCode);
-            console.log('headers:', res.headers);
-        
-            res.on('data', (d) => {
-                process.stdout.write(d);
-            });
+          let data = '';
+          res.on('data', (chunk) => {
+            data += chunk;
+          });
+          
+          res.on('end', () => {
+            if (res.statusCode >= 200 && res.statusCode < 300) {
+              console.log('Request successful');
+            } else {
+              console.error('Request failed with status:', res.statusCode);
+            }
+          });
         });
-        
+
         req.on('error', (error) => {
-            console.error("Error sending error to telemetry", error);
+          console.error('Error:', error.message);
+          process.exit(1);
```

Comment:
Using `process.exit(1)` in error handling is too aggressive and could lead to unexpected termination of the application. Consider using a callback or event emitter pattern to handle errors more gracefully. This would allow the parent application to decide how to handle failures.